controller/src/main/java/io/pravega/controller/server/rest/resources/StreamMetadataResourceImpl.java

```diff
@@ -407,4 +412,51 @@ public void updateStreamState(final String scopeName, final String streamName,
         }).thenApply(asyncResponse::resume)
                 .thenAccept(x -> LoggerHelpers.traceLeave(log, "updateStreamState", traceId));
     }
+
+    /**
+     * Implementation of getScalingEvents REST API.
+     *
+     * @param scopeName         The scope name of stream.
+     * @param streamName        The name of stream.
+     * @param from              DateTime from which scaling events should be displayed.
+     * @param to                DateTime until which scaling events should be displayed.
+     * @param securityContext   The security for API access.
+     * @param asyncResponse     AsyncResponse provides means for asynchronous server side response processing.
+     */
+    @Override
+    public void getScalingEvents(final String scopeName, final String streamName, final Long from, final Long to,
+                                 final SecurityContext securityContext, final AsyncResponse asyncResponse) {
+        long traceId = LoggerHelpers.traceEnter(log, "getScalingEvents");
+
+        controllerService.getScaleRecords(scopeName, streamName).thenApply(listScaleMetadata -> {
```

Comment:
We are getting all records and filtering everytime, can we instead pass on the time range params into the controllerservice function and filter at the store layer? only if its possible to filter internally to save I/O operations.
Also if this list can potentially be large, then we may need some limit or paging support, file an issue for this if this is a valid case.